Chore/not allowed functions

Jakobus42 opened the pull request:
- Removed/Replaced not allowed functions.
- added a SUBMISSION_BUILD macro to exclude not allowed functions from the build. (currently only used for fcntl)

Closes #58

rwxg commented on include/shared/platform.hpp:
```diff
@@ -0,0 +1,17 @@
+#pragma once
+
+#if defined(__APPLE__)
+#	define PLATFORM_HAS_KQUEUE 1
+#elif defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__)
+#	define PLATFORM_HAS_KQUEUE 1
+#elif defined(__linux__)
+#	define PLATFORM_HAS_EPOLL 1
+#else
+#	error "unsupported platform"
+#endif // platform check
+
+#if defined(SUBMISSION_BUILD)
+#	define ALLOW_FORBIDDEN_FUNCTIONS 0
+#else
+#	define ALLOW_FORBIDDEN_FUNCTIONS 1
+#endif // submission check
```
Neat.

rwxg reviewed (approved):
Good stuff, looks alright. I remember there being another disallowed function somewhere, can't for the life of me remember where or which one it was though.

Jakobus42 commented:
> Good stuff, looks alright. I remember there being another disallowed function somewhere, can't for the life of me remember where or which one it was though.

I am pretty sure I removed all of them - maybe we got rid of it in the refactor or smth.
checked with nm -u so I am fairly comfortable that I removed all of them.
